lib/pxconfig.js

```diff
@@ -36,16 +36,28 @@ class PxConfig {
             ['MODULE_MODE', 'moduleMode'], ['FIRST_PARTY_ENABLED', 'firstPartyEnabled'], ['ADDITIONAL_ACTIVITY_HANDLER', 'additionalActivityHandler'], ['ENRICH_CUSTOM_PARAMETERS', 'enrichCustomParameters'],
             ['TESTING_MODE', 'testingMode'], ['WHITELIST_EXT', 'whitelistExt'], ['BYPASS_MONITOR_HEADER', 'bypassMonitorHeader'], ['ADVANCED_BLOCKING_RESPONSE', 'advancedBlockingResponse'],
             ['TELEMETRY_COMMAND_HEADER', 'telemetryCommandHeader'], ['CUSTOM_TEMPLATE_ROOT', 'customTemplateRoot'], ['CUSTOM_TEMPLATE_DATA', 'customTemplateData'], ['FILTER_BY_IP', 'filterByIP'], ['FILTER_BY_USERAGENT', 'filterByUserAgent'], ['FILTER_BY_METHOD', 'filterByMethod'],
-            ['EXTERNAL_ACTIVITIES', 'externalActivities'], ['PXHD_SECURE', 'pxhdSecure']];
+            ['EXTERNAL_ACTIVITIES', 'externalActivities'], ['PXHD_SECURE', 'pxhdSecure'], ['BACKEND_URL', 'backendUrl']];
         
         configKeyMapping.forEach(([targetKey, sourceKey]) => {
             this.PX_DEFAULT[targetKey] = PxConfig.configurationsOverriding(this.PX_DEFAULT, params, targetKey, sourceKey);
         });
 
-        this.PX_INTERNAL.SERVER_HOST = this.PX_DEFAULT.PX_APP_ID !== 'PX_APP_ID' ? this.PX_INTERNAL.SERVER_HOST = 'sapi-' + this.PX_DEFAULT.PX_APP_ID.toLowerCase() + '.perimeterx.net' : 'sapi.perimeterx.net';
+
+        // set server host
+        if(this.PX_DEFAULT['BACKEND_URL']) {
+            this.PX_INTERNAL.SERVER_HOST_SCHEME = this.PX_DEFAULT['BACKEND_URL'].startsWith('http://') ? 'http' : 'https';
+            this.PX_INTERNAL.SERVER_HOST = this.PX_DEFAULT['BACKEND_URL'].replace(/^(https:\/\/)|^(http:\/\/)/,"");
+        }
+        else {
+            this.PX_INTERNAL.SERVER_HOST_SCHEME = 'https'
+            this.PX_INTERNAL.SERVER_HOST = this.PX_DEFAULT.PX_APP_ID !== 'PX_APP_ID' ? this.PX_INTERNAL.SERVER_HOST = 'sapi-' + this.PX_DEFAULT.PX_APP_ID.toLowerCase() + '.perimeterx.net' : 'sapi.perimeterx.net';
```

Comment:
so this is the default sever url? it supposed to be in pxDefaultConfig. not hard coded here.